tokio-postgres/src/lib.rs

```diff
@@ -246,8 +246,31 @@ pub enum SimpleQueryMessage {
     Row(SimpleQueryRow),
     /// A statement in the query has completed.
     ///
-    /// The number of rows modified or selected is returned.
-    CommandComplete(u64),
+    /// The command tag and number of rows modified or selected is returned.
+    CommandComplete(CommandTag),
+}
+
+/// Command Tag returned by the `SimpleQuery` stream.
+#[derive(Debug)]
+pub struct CommandTag(String);
+
+impl CommandTag {
+    pub(crate) fn new(tag: impl ToString) -> Self {
+        Self(tag.to_string())
+    }
+
+    /// The number of affected rows
+    pub fn rows(&self) -> u64 {
+        self.0
+            .rsplit(' ')
+            .next()
+            .map_or(0, |w| w.parse().unwrap_or(0))
+    }
```

Comment:
Remember that you're changing tokio-postgres here and when we try to get this code back upstream (might not happen because `pgcat`) they're highly likely to reject this change.

They had a nice data type for reporting the number of rows change, _correctly_ typed (using a number for an actual number), and we've changed it to be a stringly-typed.

The change will also cause a row count to *always* be reported (even if it's not relevant) because it falls back to `0`.

Seems wrong.

There might be a legitimate reason to change the type of the argument - but it's not obvious without explanation or context.